Added alias (Path) to the FilePath parameter of the Out-File cmdlet.

Noutlaw1 opened the pull request:
Relevant issue: https://github.com/PowerShell/PowerShell/issues/2923

I'm new to Github beyond using it to store some personal projects, so, if I did something wrong, sorry and just let me know! It seemed a pretty easy one for a newbie to change without accidentally breaking anything else.

I made sure to build and test it beforehand:
![out-file](https://cloud.githubusercontent.com/assets/4411919/21590142/f31e1270-d0c6-11e6-903c-22dfdaf04b49.PNG)

Noutlaw1 committed:
Added parameter alias to the Out-File cmdlet, allowing -FilePath to be used as -Path.

Relevant issue: https://github.com/PowerShell/PowerShell/issues/2923

msftclas commented:
Hi __@Noutlaw1__, I'm your friendly neighborhood Microsoft Pull Request Bot (You can call me MSBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, MSBOT;

iSazonov commented:
@Noutlaw1 Thanks for your contribution!
You should place your tests in[ this file](https://github.com/PowerShell/PowerShell/blob/master/test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1).

Noutlaw1 committed:
Test to confirm the alias I added to the Out-File cmdlet's -FilePath parameter works.

Noutlaw1 commented:
@iSazonov Thanks! I've never used Pester before but it's pretty cool!

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,11 @@ Describe "Out-File" -Tags "CI" {
 	Set-ItemProperty -Path $testfile -Name IsReadOnly -Value $false
     }
+    It "Creates a file using Out-File with the alias 'Path' for the -filepath parameter" {
+    {Out-File -path .\test.txt | Should Not Throw}
```
The 'describe' block already define '$testfile' as test file path and define 'AfterEach' block to remove a test file after each test. So your test would be:
```powershell
    It "Should be able to use 'Path' alias for 'Filepath' parameter" {
	{ Out-File -Path $testfile -InputObject $expectedContent } | Should Not Throw
    }
```

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,11 @@ Describe "Out-File" -Tags "CI" {
 	# reset to not read only so it can be deleted
 	Set-ItemProperty -Path $testfile -Name IsReadOnly -Value $false
     }
+    It "Creates a file using Out-File with the alias 'Path' for the -filepath parameter" {
```
It would be good to use the header format that is used throughout the test file.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,11 @@ Describe "Out-File" -Tags "CI" {
+    It "Creates a file using Out-File with the alias 'Path' for the -filepath parameter" {
+    {Out-File -path .\test.txt | Should Not Throw}
+    
+    #Remove the test file it created.
```
Comments are very welcome, but obvious comments are not needed.

Noutlaw1 committed:
Changed test for parameter alias as per feedback.

iSazonov commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,10 @@ Describe "Out-File" -Tags "CI" {
 	Set-ItemProperty -Path $testfile -Name IsReadOnly -Value $false
     }
+    It "Should be able to use the 'Path' alias for the 'FilePath' parameter" {
+    {Out-File -Path $testfile -InputObject $expectedContent | Should Not Throw}
```
Right pattern here is `{ #your code } | Should Not Throw`.
You can find multiple samples in the same Out-File.Tests.ps1 file.
(`Describe`, `Context`, `It` and `Should` is Pester's functions. So you send your test code to a Pester function as script block parameter, Pester execute it and piping _result of your script block_ to `Should` to check result - ok or fail.)

Noutlaw1 replied:
Made the changes. Thanks for the feedback.

mirichmo commented:
I restarted the Travis job since it looked like it hung

Noutlaw1 committed:
Updating based on feedback.

Made formatting changes:
Added space between brackets and start of cmdlet.
Moved end bracket to before the pipe and Should Not Throw statement.
Removed -InputObject parameter for the test, as it was not needed.

mirichmo commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,9 @@ Describe "Out-File" -Tags "CI" {
 	# reset to not read only so it can be deleted
```
You didn't cause it, but since you are modifying the file, I have a small refactoring request for you. Please place lines 21-23 in a BeforeAll block like this:
``` powershell
    BeforeAll {
        $expectedContent = "some test text"
        $inObject = New-Object psobject -Property @{text=$expectedContent}
        $testfile = Join-Path -Path $TestDrive -ChildPath outfileTest.txt
    }
````

mirichmo commented on test/powershell/Modules/Microsoft.PowerShell.Utility/Out-File.Tests.ps1:
```diff
@@ -127,4 +127,9 @@ Describe "Out-File" -Tags "CI" {
 	Set-ItemProperty -Path $testfile -Name IsReadOnly -Value $false
     }
+    
+	It "Should be able to use the 'Path' alias for the 'FilePath' parameter" {
```
Please fix the indentation here and make sure that you are not using tabs.

mirichmo reviewed (changes requested):
Thanks for the contribution. The code looks good and I only have two small requests.

Noutlaw1 committed:
Added BeforeAll to the first part of the file. Replaced tabs with spaces.

#2945

Noutlaw1 commented:
Sorry about the delay, I've been pretty sick lately. I made the changes you requested. Thanks for the feedback!

mirichmo committed:
Fixing indentation within the test file

mirichmo commented:
Your changes look good. I wanted to experiment with updating a PR and thought that this would be a safe case (fixing whitespace indentation), so that is why the additional commit appeared. I'll merge it once CI passes.
